src/foam/nanos/auth/UserAndGroupAuthService.js

```diff
@@ -65,6 +66,32 @@ foam.CLASS({
   ],
 
   methods: [
+    {
+      name: 'authorizeVacancy',
+      documentation: `
+        Authorizes a vacant user that has no true ownership. The assigned vacant user is relative to a spid,
+        holding various permissions allowing a user who has not logged into the system to interact with it as if they had.
+      `,
+      javaCode: `
+        ServiceProvider serviceProvider = (ServiceProvider) ((DAO) x.get("localServiceProviderDAO")).find((String) x.get("spid"));
+        if ( serviceProvider == null ) {
+          throw new AuthorizationException("Service Provider doesn't exist. Unable to authorize vacant user.");
+        }
+
+        User vacantUser = (User) ((DAO) x.get("localUserUserDAO")).find(serviceProvider.getVacantUser());
+        if ( vacantUser == null ) {
+          throw new AuthorizationException("Unable to find vacant user.");
+        }
+        x = x.put("subject", new Subject.Builder(x).setUser(vacantUser).setVacantMode(true).build());
+
+        Session session = x.get(Session.class);
+        if ( session.getUserId() == vacantUser.getId() ) return ((Subject) x.get("subject"));
+        session.setUserId(vacantUser.getId());
```

Comment:
what happens if this is called when there is a logged in user? Are we letting them proceed with this?